src/cluster/cluster.go

```diff
@@ -1,90 +1,123 @@
 package cluster
 
 import (
-	"firecontroller/microcontroller"
-	mc "firecontroller/microcontroller"
-	"firecontroller/utilities"
+	device "devicecommander/device"
+	"errors"
 	"log"
+	"net/http"
+	"strconv"
+	"time"
 
 	"github.com/spf13/viper"
 )
 
-//Cluster - This object defines an array of microcontrollers
+//Cluster - This object defines an array of Devices
 type Cluster struct {
-	Name             string
-	Microcontrollers []mc.Microcontroller
-	Me               *mc.Microcontroller
+	Name    string
+	Devices []device.Device
 }
 
-//Config -
-type Config struct {
-	Name             string `yaml:"Name"`
-	Microcontrollers []mc.Config
-}
-
-//GetConfig -
-func (c Cluster) GetConfig() (config Config) {
-	config.Name = c.Name
-	config.Microcontrollers = make([]mc.Config, len(c.Microcontrollers))
-	for i, micro := range c.Microcontrollers {
-		config.Microcontrollers[i] = micro.GetConfig()
+//GetDevices returns a map of all registered
+func (c Cluster) GetDevices() map[int]device.Device {
+	micros := make(map[int]device.Device)
+	for i := 0; i < len(c.Devices); i++ {
+		micros[c.Devices[i].ID] = c.Devices[i]
 	}
-
-	return
+	return micros
 }
 
-//Master - returns a pointer to the Master micro
-func (c *Cluster) Master() *mc.Microcontroller {
-	for _, micro := range c.Microcontrollers {
-		if micro.Master {
-			return &micro
+//AddDevice attempts to add a device to the cluster and returns the response data. This should only be run by the master.
```

Comment:
What do you mean by master?

Reply:
Nada, it's a missed GoFire comment. Will delete

Reply:
```suggestion
//AddDevice attempts to add a device to the cluster and returns the response data.